### PR TITLE
Use cache for jest

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Add jest cache across the CI to improve jest execution time.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -254,7 +254,8 @@
             "text-summary",
             "lcov"
         ],
-        "testResultsProcessor": "@casualbot/jest-sonar-reporter"
+        "testResultsProcessor": "@casualbot/jest-sonar-reporter",
+        "cacheDirectory": "/tmp/jest_cache"
```
Comment (t3chguy) [inline package.json:258]: This doesn't exist on Windows so will likely cause issues

Comment (florianduros) [reply package.json:258]: It's an attempt to see if the CI is running faster with jest cache. If there is improvement in execution time, I'll see how make it works an all platforms

Comment (t3chguy) [reply package.json:258]: Can probably be passed as a CLI option specifically in the CI workflow ftr

Comment (florianduros) [reply package.json]: Moved it to the CI workflow

**Review by andybalaam — APPROVED** [review]
Looks good; thank you for looking into this.

**Pull request merged**